src/SDK/Trace/SpanBuilder.php

```diff
@@ -36,24 +34,39 @@ final class SpanBuilder implements API\SpanBuilderInterface
      */
     private int $spanKind = API\SpanKind::KIND_INTERNAL;
 
-    /** @var list<LinkInterface>|null */
-    private ?array $links = null;
+    /** @var list<LinkInterface> */
+    private array $links = [];
 
-    private ?AttributesInterface $attributes = null;
-    private int $totalNumberOfLinksAdded = 0;
-    private int $startEpochNanos = 0;
+    private AttributesBuilderInterface $attributes;
+    private int $droppedLinksCount;
+    private int $eventCountLimit;
+    private ?int $startEpochNanos = null;
 
     /** @param non-empty-string $spanName */
     public function __construct(
         string $spanName,
         InstrumentationLibrary $instrumentationLibrary,
-        TracerSharedState $tracerSharedState,
-        SpanLimits $spanLimits
+        ResourceInfo $resource,
+        SamplerInterface $sampler,
+        SpanProcessorInterface $spanProcessor,
+        IdGeneratorInterface $idGenerator,
+        AttributesBuilderInterface $attributes,
+        AttributesFactoryInterface $linkAttributes,
+        AttributesFactoryInterface $eventAttributes,
```

Comment:
It's not clear when Attributes are the Builder and when they are Factory. Probably the proper naming of the arguments and properties could help with it.